pyarr/readarr.py

```diff
@@ -0,0 +1,553 @@
+        return self.request_get(path, self.ver_uri)
+
+    # GET /delayprofile
+    def get_delay_profiles(self):
+        """Gets all delay profiles
+
+        Returns:
+            JSON: Array
+        """
+        path = "/delayprofile"
+        return self.request_get(path, self.ver_uri)
+
+    # GET /releaseprofile
+    def get_release_profiles(self):
+        """Gets all release profiles
+
+        Returns:
+            JSON: Array
+        """
+        path = "releaseprofile"
+        return self.request_get(path, self.ver_uri)
+
+    ## BOOKS
+    # GET /book and /book/{id}
+    def get_book(self, id_=None):
+        """Returns all books in your collection or the book with the matching
+        book ID if one is found.
+
+        Args:
+            id_ (int, optional): Database id for book. Defaults to None.
+
+        Returns:
+            JSON: Array
+        """
+        path = f"book/{id_}" if id_ else "book"
+        return self.request_get(path, self.ver_uri)
+
+    # POST /book
+
+    def add_book(
+        self,
+        db_id,
+        book_id_type,
+        root_dir,
+        quality_profile_id=1,
+        metadata_profile_id=0,
+        monitored=True,
+        search_for_new_book=False,
+        author_monitor="all",
+        author_search_for_missing_books=False,
+    ):
+        """Adds a new book and  its associated author (if not already added)
+
+        Args:
+            db_id (int): goodreads, isbn, asin ID for the book
+            book_id_type (str): goodreads / isbn / asin
+            root_dir (str): Directory for book to be stored
+            quality_profile_id (int, optional): quality profile id. Defaults to 1.
+            metadata_profile_id (int, optional): metadata profile id. Defaults to 0.
+            monitored (bool, optional): should the book be monitored. Defaults to True.
+            search_for_new_book (bool, optional): search for the book to download now. Defaults to False.
+            author_monitor (str, optional): monitor the author for new books. Defaults to "all".
+            author_search_for_missing_books (bool, optional): search for missing books from this author. Defaults to False.
+
+        Raises:
+            ValueError: error raised if book_id_type is incorrect
+
+        Returns:
+            JSON: Array
+        """
+        book_id_types = ["goodreads", "isbn", "asin"]
+        if book_id_type not in book_id_types:
+            raise ValueError(f"Invalid book id type. Expected one of: {book_id_types}")
+
+        book_json = self._construct_book_json(
+            db_id,
+            book_id_type,
+            root_dir,
+            quality_profile_id,
+            metadata_profile_id,
+            monitored,
+            search_for_new_book,
+            author_monitor,
+            author_search_for_missing_books,
+        )
+        path = "book"
+        return self.request_post(path, self.ver_uri, data=book_json)
+
+    # PUT /book
+    def upd_book(self):
```

Comment:
This either needs to be developed or fully commented out

Reply:
This was duplicated, now removed